### PR TITLE
Introducing an INTERPRET-Inspired Baseline for VLM predicate

### PR DESCRIPTION
**Pull request by yichao-liang**
This baseline assumes a 100% accurate vision-API return.

This can be used with the `--vlm_predicate_vision_api_generate_ground_atoms True` flag.

Example usage:
```
xvfb-run -a python predicators/main.py --env kitchen --approach grammar_search_invention --seed 0 --num_train_tasks 1 --num_test_tasks 1 --bilevel_plan_without_sim True --perceiver kitchen --offline_data_method demo_with_vlm_imgs --vlm_model_name gemini-1.5-pro-latest --segmenter every_step --grammar_search_vlm_atom_proposal_prompt_type options_labels_whole_traj --grammar_search_task_planning_timeout 10.0 --sesame_max_skeletons_optimized 200 --excluded_predicates all --disable_harmlessness_check True --debug --kitchen_goals boil_kettle --kitchen_use_perfect_samplers True --grammar_search_vlm_atom_proposal_use_debug False --excluded_predicates all --make_failure_videos --kitchen_use_combo_move_nsrts True --kitchen_randomize_init_state True --kitchen_render_set_of_marks True --grammar_search_vlm_atom_label_prompt_type img_option_diffs --vlm_predicate_vision_api_generate_ground_atoms True
```

### CODE REVIEW

File: predicators/datasets/generate_atom_trajs_with_vlm.py [hunk 1]
```diff
@@ -735,6 +739,178 @@ def _query_vlm_to_generate_ground_atoms_trajs(
...
+        response = vlm.sample_completions(prompt,
+                                          imgs,
+                                          0.0,
+                                          CFG.seed,
+                                          num_completions=1)[0]
+        response_file = prompt_dir + "response.txt"
+        with open(response_file, 'w', encoding="utf-8") as f:
+            f.write(response)
+        # 3. Parse the responses into a set of predicates
+        candidates |= _parse_predicate_proposals(response_file, train_tasks,
```
Comment (NishanthJKumar) [inline predicators/datasets/generate_atom_trajs_with_vlm.py:767]: hold on - why are you writing the response to a file like this? Don't we already cache responses by default?

File: predicators/datasets/vlm_input_data_prompts/vision_api/response.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+```python
```
Comment (NishanthJKumar) [inline predicators/datasets/vlm_input_data_prompts/vision_api/response.txt:1]: given this file is dynamically generated, could you add it to the gitignore? (unless you decide to get rid of it given the above comment)

File: predicators/datasets/generate_atom_trajs_with_vlm.py [hunk 3]
```diff
@@ -816,9 +992,12 @@ def create_ground_atom_data_from_generated_demos(
     # Now, given these trajectories, we can query the VLM.
     if vlm is None:
         vlm = utils.create_vlm_by_name(CFG.vlm_model_name)  # pragma: no cover
-    ground_atoms_trajs = _query_vlm_to_generate_ground_atoms_trajs(
-        img_option_trajs, env, train_tasks, known_predicates, all_task_objs,
-        vlm)
+    if CFG.vlm_predicate_vision_api_generate_ground_atoms:
+        generate_func = _generate_ground_atoms_trajs_from_synthesized_predicates
+    else:
+        generate_func = _query_vlm_to_generate_ground_atoms_trajs
```
Comment (NishanthJKumar) [inline predicators/datasets/generate_atom_trajs_with_vlm.py:998]: NIT: could you rename `_generate_ground_atoms_from_synthesized_predicates` to `_generate_ground_atoms_with_vlm_oo_code_gen`, and rename `_query_glm_to_generate_ground_atoms_trajs` to `_generate_ground_atoms_with_vlm_pure_visual_preds`?

File: predicators/datasets/vlm_input_data_prompts/vision_api/types_burger.txt [hunk 4]
```diff
@@ -0,0 +1,16 @@
+_object_type = Type("object", [])
```
Comment (NishanthJKumar) [inline predicators/datasets/vlm_input_data_prompts/vision_api/types_burger.txt:1]: I think this way of doing things is fine, but I wonder if we could dynamically generate this text from our types in every env automatically? Did you try this/do you think it's possible? That way, we wouldn't have to create a new file like this for every env/change it whenever the env changes.

**Review by NishanthJKumar — COMMENTED** [review]
Great work overall; thanks so much for doing this so quickly! Left a couple questions/points of feedback. Feel free to respond here or on slack!

File: predicators/datasets/generate_atom_trajs_with_vlm.py (hunk 1 above)
Comment (yichao-liang) [reply predicators/datasets/generate_atom_trajs_with_vlm.py:767]: It was just for convenience. I'll remove it.

**Review by NishanthJKumar — APPROVED** [review]
LGTM! Feel free to merge once all the checks pass 💯

**Pull request merged**